### PR TITLE
Hclang 63 create more tests for lexer

### PR DESCRIPTION
**Pull request by Lambcreek**
### Description
More test methods for Lexer. Some tests are currently ignored, as they fail and can't build

### Changelog
 - Added tests that make sure the lexer correctly creates tokens

### Issues solved 
 - HCLANG-63

## FOR REVIEWER [Do not remove]
A small checklist of things to note when you are doing a review. Make sure these things apply before going into master.
 - Did the PR actually fix the issue
 - Comment anything that doesn't make sense or anything you want clarified. Many comments are always ok
 - Make sure this is ready for exams if this is going into master
 - It's always a good idea having another reviewer look at a PR as well, if it's major. get a buddy.

### Regarding filenames
- all .tex files use pascal case
- all other files [except in /code/] use snake case

### Regarding code [if applicable]
- Check [coding conventions](https://kotlinlang.org/docs/reference/coding-conventions.html)
- Are public methods documented? 
- Are there testcases for new features / methods?
- Have you tried running the source and tested things that cannot be test-cased?

### Regarding Report [if applicable]
- Does each line have a linechange in the sourcecode?
- Does sentences make sense?
- Did you do spellchecking?
- Citations?

### CODE REVIEW

File: code/src/main/kotlin/lexer/Lexer.kt [hunk 1]
```diff
@@ -7,7 +7,7 @@ import kotlin.coroutines.experimental.buildSequence
  */
 class Lexer(private val inputContent: String) : ILexer {
     private val endLines = listOf("\n","\r\n",";")
-    private val specialChars = listOf("=", "[", "]", "(", ")", "{", "}", ",", "\\", "->", "\\") + endLines
+    private val specialChars = listOf("=", "[", "]", "(", ")", "{", "}", ",", "\\", ":", "\\") + endLines
```
Comment (nillas12) [inline code/src/main/kotlin/lexer/Lexer.kt]: Nice!

Comment (C0DK) [reply code/src/main/kotlin/lexer/Lexer.kt]: this has been fixed multiple places :P prolly gon get merge conflicts but ohwell

File: code/src/test/kotlin/lexicalTests/LexerTest.kt [hunk 2]
```diff
@@ -76,6 +74,176 @@ class LexerTest {
...
+
+        val tokens = lex.getTokenSequence().toList()
+
+        assertTrue(tokens[0].token is lexer.Token.Literal.Bool)
+        assertTrue(tokens[1].token is lexer.Token.Literal.Bool)
+    }
+
+    @org.junit.jupiter.api.Test
+    fun lexerTestFunctionDeclaration() {
+        val lex = Lexer("func print = (txt message) : none\n{ }")
```
Comment (C0DK) [inline code/src/test/kotlin/lexicalTests/LexerTest.kt]: we need to test that =( is two symbols as well. maybe in general that no-white-space works with special chars in combo with both identifiers and types - there is a method that does this for identifiers already. that robably just needs to be extended

File: code/src/test/kotlin/lexicalTests/LexerTest.kt [hunk 3]
```diff
@@ -76,6 +74,176 @@ class LexerTest {
...
+                { token -> token is lexer.Token.SpecialChar.BlockEnd },
+                2, 1)
+        assertPositionalToken(tokens[12],
+                { token -> token is lexer.Token.SpecialChar.EndOfLine },
+                3, 1)
+    }
+
+    @org.junit.jupiter.api.Test
+    fun lexerTestDoesntRequireWhitespaces() {
+        val lex = Lexer("x=y[0]")
```
Comment (C0DK) [inline code/src/test/kotlin/lexicalTests/LexerTest.kt]: test this with types followed by special char as well (as previously commented..) something like "(txt" and "=(" - both cases needs testing

File: code/src/test/kotlin/lexicalTests/LexerTest.kt [hunk 4]
```diff
@@ -76,6 +74,176 @@ class LexerTest {
...
+                4, 0)
+
+        assertPositionalToken(tokens[5],
+                { token -> token is lexer.Token.SpecialChar.SquareBracketEnd },
+                5, 0)
+    }
+
+    @org.junit.jupiter.api.Test
+    fun lexerTestInlineLambdaExpression() {
+        val lex = Lexer("x = 1 to 10 map {value * 2}")
```
Comment (C0DK) [inline code/src/test/kotlin/lexicalTests/LexerTest.kt]: is this test any different that the func assignment? this seems more relevant in the parser - this doesn't really have any "unique" cases that the others did not cover. So it's just going to make building slower without adding anything new. maybe just have a tests for ALL symbols (as you already have for all types)

**Pull request merged**